package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -104,6 +105,15 @@ define([
       this.__model = model;
 
       /**
+       * The state of the view.
+       *
+       * @type {object}
+       * @readOnly
+       * @private
+       */
+      this.__state = state;
```

Comment:
I don't think state should be stored by the base class — it is "abstract" in this respect.
It is up to specific view sub-classes to "parse" it and store it somehow.